### DISCUSSION TITLE
Presence channel does not work if wire:navigate 'd but working if the page is refreshed

### DISCUSSION THREAD

**Discussion by thenibirahmed** [livewire/livewire#8854] [1. Help] [answered]
I'm just showing the name of the users in a presence channel. It's working fine on refresh or without wire:navigate but doesn't work if wire:navigated

please help

```php
@foreach ($users as $user)
	<img class="inline-block h-6 w-6 rounded-full ring-2 ring-white" src="https://ui-avatars.com/api/?name={{ $user['name'] ?? '' }}">
@endforeach
```

```php
public function getListeners()
    {
        return [
            "echo-presence:audiences,here" => 'handleAudiences',
            "echo-presence:audiences,joining" => 'addAudience',
            "echo-presence:audiences,leaving" => 'removeAudience',
        ];
    }
```

**Comment by sinnbeck** [comment]
Dont suppose you found a solution to this? I am facing the same issue

**Comment by lrljoe** [comment]
I tend to handle all of that client side, what are you needing to do server side with those events?

**Comment by thenibirahmed** [comment, marked as answer]
After deep analysis, I found that, the problem is when we wire:navigate the first time it subscribes to the channel. Then livewire doesn't leave the channel when we go to another page. That's why it doesn't work before refresh. So we need to make the app leave the channel as soon as we move to another page or get away from the component.

I fixed it by adding this line of code of alpine in the parent div. Everything is same other than this.

```php
x-data="{
     destroy() {
	  Echo.leave('my-channel.{{ $model->id }}');
     },
}"
```

Hope it helps
